MdeModulePkg/Universal/HiiDatabaseDxe/Database.c

```diff
@@ -1057,23 +1073,29 @@ UpdateDefaultSettingInFormPackage (
           Width = (UINTN)((UINT32)1 << (((EFI_IFR_ONE_OF *)IfrOpHdr)->Flags & EFI_IFR_NUMERIC_SIZE));
         }
 
-        EfiVarStoreIndex     = IsEfiVarStoreQuestion (IfrQuestionHdr, EfiVarStoreList, EfiVarStoreNumber);
         StandardDefaultIsSet = FALSE;
         ManufactDefaultIsSet = FALSE;
+        if (EfiVarStoreList == NULL) {
+          goto Done;
+        }
+
+        EfiVarStoreIndex = IsEfiVarStoreQuestion (IfrQuestionHdr, EfiVarStoreList, EfiVarStoreNumber);
         //
         // Find Default and Manufacturing default for OneOf question
         //
         if (EfiVarStoreIndex < EfiVarStoreNumber) {
           for (Index = 0; Index < DefaultIdNumber; Index++) {
-            if (DefaultIdList[Index] == EFI_HII_DEFAULT_CLASS_STANDARD) {
-              Status = FindQuestionDefaultSetting (EFI_HII_DEFAULT_CLASS_STANDARD, EfiVarStoreList[EfiVarStoreIndex], IfrQuestionHdr, &IfrValue, Width, QuestionReferBitField);
-              if (!EFI_ERROR (Status)) {
-                StandardDefaultIsSet = TRUE;
-              }
-            } else if (DefaultIdList[Index] == EFI_HII_DEFAULT_CLASS_MANUFACTURING) {
-              Status = FindQuestionDefaultSetting (EFI_HII_DEFAULT_CLASS_MANUFACTURING, EfiVarStoreList[EfiVarStoreIndex], IfrQuestionHdr, &IfrManufactValue, Width, QuestionReferBitField);
-              if (!EFI_ERROR (Status)) {
-                ManufactDefaultIsSet = TRUE;
+            if (DefaultIdList != NULL) {

```

Comment:
Should we add an ASSERT here:
```
if (DefaultIdList == NULL) {
  ASSERT (DefaultIdList != NULL;
} else {
```

Reply:
I'm adding a break as well.

ASSERT by itself is not enough to cover cases. if DebugAssertEnabled returns false (`PcdDebugPropertyMask has bit DEBUG_PROPERTY_DEBUG_ASSERT_ENABLED cleared) , code execution will proceed pass the assert